misc/test_runner/test_runner.c

```diff
@@ -84,7 +84,21 @@ int main() {
    * invocation of the Test macro
    */
   sigaction(SIGSEGV, &act, NULL);
+
+  // Reverse tests, as the `__attribute__((constructor))` approach with a linked list makes them backwards.
+  size_t i = 0;
+  for (struct fake_criterion_test *test_info = fake_criterion_tests; test_info; test_info = test_info->next) {
+      i++;
+  }
+  const size_t num_tests = i;
+  struct fake_criterion_test tests[num_tests];
```

Comment:
I do cringe whenever I see a VLA but this should be fine in practice since this is just test code.